### PR TITLE
[dashboards] New, tittle and slug OR filter

### PR DESCRIPTION
**Pull request by dpgaspar**
### CATEGORY

- [ ] Bug Fix
- [X] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
Add a custom filter to dashboards so that it's possible to filter by dashboard tittle and slug at the same time. This is a requirement for future dashboard list view design. Requires FAB 2.3.2 still on RC1

Needs this first:
#9491

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@nytai

### CODE REVIEW

File: superset/dashboards/filters.py [hunk 1]
```diff
@@ -23,6 +24,19 @@
...
 
 
+class DashboardTitleOrSlugFilter(BaseFilter):  # pylint: disable=too-few-public-methods
+    name = _("Title or Slug")
+    arg_name = "title_or_slug"
+
+    def apply(self, query, value):
+        return query.filter(
+            or_(
+                Dashboard.dashboard_title.ilike(value + "%"),
```
Comment (willbarrett) [inline superset/dashboards/filters.py:34]: Should there be `%` at the beginning of these filters as well?

Comment (dpgaspar) [reply superset/dashboards/filters.py:34]: Good idea, changed it

File: setup.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ def get_git_sha():
         "croniter>=0.3.28",
         "cryptography>=2.4.2",
         "flask>=1.1.0, <2.0.0",
-        "flask-appbuilder>=2.3.1, <2.4.0",
+        "flask-appbuilder==2.3.2rc1",
```
Comment (nytai) [inline setup.py:79]: nice, this makes more sense

Comment (dpgaspar) [reply setup.py:79]: I actually want to revert this before changing and after flask-appbuolder 2.3.2 is out

File: superset/dashboards/filters.py [hunk 3]
```diff
@@ -23,6 +24,20 @@
 from superset.views.base import BaseFilter, get_user_roles
 
 
+class DashboardTitleOrSlugFilter(BaseFilter):  # pylint: disable=too-few-public-methods
+    name = _("Title or Slug")
+    arg_name = "title_or_slug"
+
+    def apply(self, query, value):
+        ilike_value = f"%{value}%"
```
Comment (nytai) [inline superset/dashboards/filters.py:32]: we may want to handle `None` or `''` cases here and not apply the filter. I think running an `ilike %%` can be very slow on some dbs. I can handle these cases before issuing the request, however we may want some protection via the api.

Comment (dpgaspar) [reply superset/dashboards/filters.py:32]: Good point!

**Review by nytai — APPROVED** [review]
just one comment, not blocking. We can address it in a followup pr if it does turn out to be an issue

**Review by willbarrett — APPROVED** [review]

**Pull request merged**